pkg/storage/ingest/reader.go

```diff
@@ -137,7 +143,14 @@ func collectFetchErrs(fetches kgo.Fetches) (_ error) {
 }
 
 func (r *PartitionReader) enqueueCommit(fetches kgo.Fetches) {
-	r.commitFetches <- fetches
+	if fetches.NumRecords() == 0 {
+		return
+	}
+	lastOffset := int64(0)
+	fetches.EachPartition(func(partition kgo.FetchTopicPartition) {
+		lastOffset = partition.Records[len(partition.Records)-1].Offset
```

Comment:
I would add an extra check on the partition, to make sure the partition matches the expected one. It should never happen, but I want to make sure we don't have a bug where we've consumed another partition and then we commit the wrong offset to the partition.